### PR TITLE
Ruff lint and format, isort, codespell

### PR DESCRIPTION
**Pull request by samaloney**
Add ruff lint and format, sort and code spell to pre-commit and make all pass

### CODE REVIEW

**Comment by settwi** [comment]
sweet!
if you run pre-commit locally does it now call ruff?

**Comment by samaloney** [comment]
> if you run pre-commit locally does it now call ruff?

Yea that's the idea, so once this is merged it should.

**Comment by samaloney** [comment]
The test fail is due to numpy2.0 removal so I think can be ignored for now

**Comment by samaloney** [comment]
Again windows test fail is unrelated - maybe getting blocked by hesperia for too many downloads

File: ruff.toml
Comment (DanRyanIrish) [inline ruff.toml:15]: What's this?

Same for lines below marked `# TODO! fix`

Comment (samaloney) [reply ruff.toml:15]: Eh ignore line too long, pytext fixture as tuples and purest raise too broad errors until they are fixed in the code base as they can't be automatically fixed

**Review by KriSun95 — APPROVED** [review]
Wow! Clearly there are quite a few words I just can't spell when I'm typing quickly! This all seems good to me.

**Comment by samaloney** [comment]
Ha I think if there was a typo award I would get it - worst thing this doesn't even catch them all.

**Pull request merged**
